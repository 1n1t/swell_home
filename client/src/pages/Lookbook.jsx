/* eslint-disable consistent-return */
import React, { Component } from 'react';
import Slider from 'react-slick';
import axios from 'axios';

class Lookbook extends Component {
	state = {
		categories: []
	};

	componentDidMount() {
		this.fetchCategories();
		this.sliderSettings.beforeChange = this.beforeSlideChange;
	}

	sliderSettings = {
		dots: true,
		infinite: true,
		speed: 500,
		slidesToShow: 1,
		slidesToScroll: 1,
		arrows: false
	};

	fetchCategories = async () => {
		const response = await axios.get('/api/categories').catch(error => error);
		if (response.data) {
			const { categories } = response.data;
			this.setState({ categories });
		}
	};

	beforeSlideChange = (oldIndex, newIndex) => {
		console.log(oldIndex, newIndex);
	};

	renderSliders = () => {
		const { categories } = this.state;

		if (categories.length > 0) {
			return categories.map(({ image, _id }) => {
				// eslint-disable-next-line import/no-dynamic-require,global-require
				const src = require(`../img/categories/${image}`);
				return (
					<div key={_id}>
						<img src={src} alt="" />
					</div>
				);
			});
		}
	};

	render() {
		return (
			<div className="Lookbook">
				<Slider className="Slider" {...this.sliderSettings}>
					{this.renderSliders()}
				</Slider>
			</div>
		);
	}
}

export default Lookbook;
