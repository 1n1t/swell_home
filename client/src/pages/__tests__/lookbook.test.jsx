import Lookbook from '../Lookbook';
import { sel, TestHelper } from '../../../../tests/utils';

const defaultProps = {};

const helper = new TestHelper(Lookbook, defaultProps);

test('the component mounts with defaults', () => {
	const wrapper = helper.mountComponent({}, true);
	expect(wrapper).toMatchSnapshot();
});

test('it knows when user changes slide', () => {
	const wrapper = helper.mountComponent();
	// console.log(wrapper.find('.slick-initialized').children().get(0));
});
