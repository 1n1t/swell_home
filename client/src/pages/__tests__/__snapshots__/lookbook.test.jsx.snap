// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the component mounts with defaults 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Lookbook />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <h1
          className="heading"
        >
          Swell.
        </h1>,
      ],
      "className": "Lookbook",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Swell.",
          "className": "heading",
        },
        "ref": null,
        "rendered": "Swell.",
        "type": "h1",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <h1
            className="heading"
          >
            Swell.
          </h1>,
        ],
        "className": "Lookbook",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Swell.",
            "className": "heading",
          },
          "ref": null,
          "rendered": "Swell.",
          "type": "h1",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
